activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -212,12 +212,12 @@ def sole
     result = nil
     found = false
 
-    each do |element|
+    each do |*element|
```

Comment:
I fear this may have a similar issue with keyword arguments, but I need to try a few things.